### PR TITLE
Always create health code

### PR DESCRIPTION
**Pull request by alxdarksage**
This started with the desire to make accounts that are created in the Stormpath UI fully editable in the Bridge Study Manager. It should also make it possible for developers we set up to use their accounts in the apps while testing, which they sometimes do. 

But then I also wanted to eliminate the code in several places to check and create the healthId/healthCode mapping if it doesn't exist. This was all highly redundant and once you ensure an account always has a health code in AccountDao, there's no reason not to return the healthCode itself as part of the account. Then it's not necessary to use healthCodeService in most places.

The iterator methods now return AccountSummary which is much faster to retrieve than an account, and then like the paged methods, you can retrieve an individual, fully-initialized account object.

In short: if you have a reference to an account it should always have a health code.

### CODE REVIEW

File: app/org/sagebionetworks/bridge/models/accounts/AccountSummary.java [hunk 1]
```diff
@@ -1,32 +1,51 @@
...
-            @JsonProperty("createdOn") DateTime createdOn, @JsonProperty("status") AccountStatus status) {
+            @JsonProperty("createdOn") DateTime createdOn, @JsonProperty("status") AccountStatus status,
+            @JsonProperty("studyIdentifier") StudyIdentifier studyIdentifier) {
         this.firstName = firstName;
         this.lastName = lastName;
         this.email = email;
         this.id = id;
         this.createdOn = (createdOn == null) ? null : createdOn.withZone(DateTimeZone.UTC);
         this.status = status;
+        this.studyIdentifier = studyIdentifier;
```
Comment (alxdarksage) [inline app/org/sagebionetworks/bridge/models/accounts/AccountSummary.java:48]: Add to SDK.

File: app/org/sagebionetworks/bridge/services/backfill/ConsentAuditAndRepairBackfill.java [hunk 2]
```diff
@@ -163,10 +168,13 @@ private long getSignedOnDate(StringBuilder sb, DateTime enrollment, long synthet
...
+     */
     private String getHealthCode(BackfillTask task, BackfillCallback callback, Account account) {
-        HealthId mapping = healthCodeService.getMapping(account.getHealthId());
-        if (mapping != null) {
-            return mapping.getCode();
+        if (account.getHealthCode() != null) {
+            return account.getHealthCode();
         }
         callback.newRecords(getBackfillRecordFactory().createOnly(task, "No health code found: " + account.getId()));
         return null;
```
Comment (alxdarksage) [inline app/org/sagebionetworks/bridge/services/backfill/ConsentAuditAndRepairBackfill.java]: Check again: this can be removed.

File: app/org/sagebionetworks/bridge/stormpath/StormpathAccountDao.java [hunk 3]
```diff
@@ -392,15 +363,33 @@ public void deleteAccount(Study study, String id) {
...
+    private Account constructAccount(StudyIdentifier studyId, com.stormpath.sdk.account.Account acct) {
+        List<SubpopulationGuid> subpopGuids = getSubpopulationGuids(studyId);
+        StormpathAccount account = new StormpathAccount(studyId, subpopGuids, acct, encryptors);
+        HealthId healthId = null;
+        if (account.getHealthCode() == null) {
+            healthId = healthCodeService.getMapping(account.getHealthId());
+        }
+        if (healthId == null) {
+            healthId = healthCodeService.createMapping(studyId);
+            account.setHealthId(healthId);
```
Comment (alxdarksage) [inline app/org/sagebionetworks/bridge/stormpath/StormpathAccountDao.java]: redundant

File: test/org/sagebionetworks/bridge/stormpath/StormpathAccountDaoMockTest.java [hunk 4]
```diff
@@ -318,28 +286,46 @@ public void updatingAccountWithRemovedGroups() {
...
+        when(mockAccount.getRoles()).thenReturn(newGroupSet);
+
+        // execute
+        dao.updateAccount(mockAccount);
+
+        // return this, so the caller can verify back-end mock calls
+        return mockSpAccount;
     }
 
 }
```
Comment (alxdarksage) [inline test/org/sagebionetworks/bridge/stormpath/StormpathAccountDaoMockTest.java]: Test that the user does NOT have the health ID in the Stormpath account. The DAO should call the healthCodeService and set this up.
